cypress/integration/add_firm.spec.js

```diff
@@ -0,0 +1,29 @@
+describe("Firm", () => {
+    beforeEach(() => {
+        cy.setCookie("Other", "other");
+        cy.setCookie("XSRF-TOKEN", "abcde");
+    });
+
+    describe("Adding a firm", () => {
+        beforeEach(() => {
+            cy.visit("/supervision/deputies/professional/deputy/1/add-firm");
+        });
+           
+        it("shows error message when submitting invalid data", () => {
+            cy.setCookie("fail-route", "firm")
+            cy.get("#add-firm-form").submit();
+            cy.get(".govuk-error-summary__title").should("contain", "There is a problem");
+            cy.get(".govuk-error-summary__list").within(() => {
+                cy.get("li:first").should("contain", "The building or street must be 255 characters or fewer");
+                cy.get("li").eq(1).should("contain", "Address line 2 must be 255 characters or fewer");
+                cy.get("li").eq(2).should("contain", "Address line 3 must be 255 characters or fewer");
+                cy.get("li").eq(3).should("contain", "The county must be 255 characters or fewer");
+                cy.get("li").eq(4).should("contain", "The email must be 255 characters or fewer");
+                cy.get("li").eq(5).should("contain", "The firm name is required and can't be empty");
+                cy.get("li").eq(6).should("contain", "The telephone number must be 255 characters or fewer");
+                cy.get("li").eq(7).should("contain", "The postcode must be 255 characters or fewer");
+                cy.get("li").eq(8).should("contain", "The town or city must be 255 characters or fewer");
+            })
+        });
+    })
+});
```

Comment:
Should there be an additional test testing you can successfully add a firm?

Reply:
done